### PR TITLE
Utiliti will now crash and generate a crash log if it fails to initialize or encounters an Error

### PR DESCRIPTION
**Pull request by Gamebuster19901**
It is possible for utiliti to have an exception get thrown before the UI is initialized. If this happens, utiliti will now crash and generate a crash report, instead of hanging and sometimes not generating a crash report. (Exception previously could get thrown before the exception handler was set).

Utiliti will also crash and generate a crash report if a subclass of Error is thrown (excluding ThreadDeath).

The previous behavior has created issues with helping game devs debug issues they are experiencing with utiliti.

see:
 https://discord.com/channels/326074836508213258/804040170504192031/826994492245540884
https://discord.com/channels/326074836508213258/804040170504192031/826992432440410143

### CODE REVIEW

File: utiliti/src/de/gurkenlabs/utiliti/Program.java [hunk 1]
```diff
@@ -17,40 +17,54 @@
...
+          Game.init(args);
+          forceBasicEditorConfiguration();
+          Game.world().camera().onZoom(event -> Editor.preferences().setZoom((float) event.getZoom()));
+    
+          // prepare UI and start the game
+          UI.init();
+          Game.start();
+        }
+        catch(Throwable t) {
+          throw new UtilitiInitializationError("Utiliti failed to initialize, see the stacktrace below for more information", t);
```
Comment (nightm4re94) [inline utiliti/src/de/gurkenlabs/utiliti/Program.java:45]: Keep the correct capitalization in mind:  **utiLITI**

**Review by nightm4re94 — CHANGES_REQUESTED** [review]
Thank you!
The only remark I have is the incorrect capitalization on utiLITI as commented. 
Cheers :)

File: utiliti/src/de/gurkenlabs/utiliti/Program.java (hunk 1 above)
Comment (Gamebuster19901) [reply utiliti/src/de/gurkenlabs/utiliti/Program.java]: Not really sure what to do with the class name, let me know if ab3fe0f is correct.

Comment (nightm4re94) [reply utiliti/src/de/gurkenlabs/utiliti/Program.java]: works for me :) Just referencing the console output previously.

**Review by nightm4re94 — APPROVED** [review]
Thank you!

**Comment by steffen-wilke** [comment]
@Gamebuster19901 Unfortunately I'm unable to start utiLITI with the changes from this PR. I had to revert the commit locally to be able to start it again. Please review the changes. It throws a `NullPointerException` when attempting to set the `DefaultUnhandledExceptionHandler` on the "gameLoop" that is not present yet since `Game.init(..)` was not called yet.

**Comment by Gamebuster19901** [comment]
My bad, it seems I forgot to rebuild the project before I tested it, so it was still running the old version.

I'll send an updated PR when I get off work.

**Comment by Gamebuster19901** [comment]
Addressed in #391

**Pull request merged**